lib/home_page.dart

```diff
@@ -0,0 +1,326 @@
+        ),
+      ),
+
+      // List of questions and a category filter
+      body: Column(
+        children: [
+          // List of Questions
+          Expanded(
+              // Show loading circle if results are taking time
+              // Show "No results" if input text doesn't match with question title (later will be added to description too)
+              child: _buildList(questionsListFiltered, filterQuestionsCategory, getCategory)),
+
+          RawMaterialButton(
+            onPressed: () {},
+            fillColor: Colors.blueGrey[800],
+            splashColor: Colors.grey,
+            child: Padding(
+              padding: EdgeInsets.all(10.0),
+              child: SizedBox(
+                width: 200.0,
+                child: Text(
+                  "Create New Question",
+                  maxLines: 1,
+                  style: TextStyle(color: Colors.white),
+                  textAlign: TextAlign.center,
+                ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+// Function to return List of Questions and category filter. Called from 'body'.
+ListView _buildList(questionsListFiltered, filterQuestionsCategory, getCategory) {
+  return ListView.builder(
+      itemCount: questionsListFiltered.length + 1,
+      itemBuilder: (context, index) {
+        if (index == 0) {
+          // category filter are above the first row. If there is no questions to show, it will still be there.
+          return Column(
+            children: <Widget>[
+              // Categories filter
+              Row(mainAxisAlignment: MainAxisAlignment.end, children: <Widget>[
+                Container(
+                    padding: EdgeInsets.symmetric(horizontal: 15),
+                    child: CategoryFilter(getCategoryCallback: getCategory))
+              ]),
+              if (questionsListFiltered.length == 0) Center(child: Text("No results"))
+            ],
+          );
+        }
+        index -= 1;
+        return Padding(
+          padding: const EdgeInsets.symmetric(vertical: 5.0, horizontal: 5.0),
+          child: Row(
+            children: [
+              Expanded(
+                child: Padding(
+                  padding: const EdgeInsets.symmetric(horizontal: 10.0),
+                  child: Column(
+                    crossAxisAlignment: CrossAxisAlignment.start,
+                    children: [
+                      Text(
+                        questionsListFiltered[index].category,
+                        textAlign: TextAlign.left,
+                        style: TextStyle(fontSize: 15.0),
+                      ),
+                      SizedBox(height: 5),
+                      Row(children: [
+                        Text(
+                          "Asked:  ",
+                          textAlign: TextAlign.left,
+                          style: TextStyle(fontSize: 15.0, color: Colors.grey),
+                        ),
```

Comment:
Here, you kinda repeat the similar code. Please make a custom Text that you can pass variables into and allow for cleaner code. Example: 
```dart
Text customText (String string, int fontSize, Colors color) {
return Text(
                          string,
                          textAlign: TextAlign.left,
                          style: TextStyle(fontSize: fontSize, color: color),
                        ),
}
```
Then you call `customText ("Asked:  ", 15.0, Colors.grey)`